### PR TITLE
Add e2e_new to gh actions

### LINKED ISSUE #872: Run `e2e_new` test suite in the E2E GH action
**Problem**

We're not running `e2e_new` test suite in the E2E GH action.

**[Persona:](https://github.com/knative/eventing/blob/master/docs/personas.md)** 
Which persona is this feature for?
- Developer

**Exit Criteria**

`e2e_new` test suite runs in the E2E GH action.

**Time Estimate (optional):** 
How many developer-days do you think this may take to resolve?
0.3

**Additional context (optional)**

https://github.com/knative-sandbox/eventing-kafka-broker/blob/f0bf0d13e42a94b94bfc1ac166dc75e176c747fe/.github/workflows/kind-e2e.yaml#L34-L52

### PR DESCRIPTION
**Pull request by slinkydeveloper**
Signed-off-by: Francesco Guardiani <francescoguard@gmail.com>

Fixes #872

## Proposed Changes

- Now GH Actions run e2e_new test suite

### CODE REVIEW

**Comment by slinkydeveloper** [comment]
I can't get why the image uploading it's not working...

```
2021/05/19 13:57:50 PWD:  /home/runner/work/eventing-kafka-broker/eventing-kafka-broker/src/knative.dev/eventing-kafka-broker/test/e2e_new
2021/05/19 13:57:50 FILENAME:  /home/runner/work/eventing-kafka-broker/eventing-kafka-broker/src/knative.dev/eventing-kafka-broker/vendor/knative.dev/reconciler-test/pkg/eventshub/resources.go
{"level":"info","ts":1621432670.4083025,"logger":"fallback","caller":"injection/injection.go:61","msg":"Starting informers..."}
level All, state Any, feature 

2021/05/19 13:57:50 failed to publish images: importpath "ko://knative.dev/reconciler-test/cmd/eventshub" is not supported: cannot find package "knative.dev/reconciler-test/cmd/eventshub" in any of:
	/opt/hostedtoolcache/go/1.15.12/x64/src/knative.dev/reconciler-test/cmd/eventshub (from $GOROOT)
	/home/runner/work/eventing-kafka-broker/eventing-kafka-broker/src/knative.dev/reconciler-test/cmd/eventshub (from $GOPATH)
error attempting to ko publish: exit status 1
```

**Comment by pierDipi** [comment]
Should we do

https://github.com/knative-sandbox/eventing-kafka-broker/blob/c1cb03b75c9704ddc5f126c7890c727a211f7687/test/e2e-common.sh#L78-L84

for rekt too? I mean the search and replace dance.

**Comment by slinkydeveloper** [comment]
@pierDipi isn't it already doing? At the end of the day, I didn't changed the scripts to run before the test... Maybe it's reconciler-test trying to push the image, while it should not...

File: .github/workflows/kind-e2e.yaml [hunk 1]
```diff
@@ -33,6 +33,7 @@ jobs:
 
         test-suite:
           - ./test/e2e
+          - ./test/e2e_new
```
Comment (pierDipi) [inline .github/workflows/kind-e2e.yaml:36]: Can we just use `./test/...` or something like that? Fewer actions consumed from the shared poll.

**Comment by pierDipi** [comment]
> isn't it already doing? 

I haven't found it

File: .github/workflows/kind-e2e.yaml (hunk 1 above)
Comment (slinkydeveloper) [reply .github/workflows/kind-e2e.yaml:36]: I'm worried there might be problems because it's going to build a single test binary...

**Comment by n3wscott** [comment]
you need to add this line: https://github.com/knative/eventing/blob/main/hack/tools.go#L34

**Comment by slinkydeveloper** [comment]
@n3wscott it's already there: https://github.com/knative-sandbox/eventing-kafka-broker/blob/main/hack/tools.go#L41

File: .github/workflows/kind-e2e.yaml (hunk 1 above)
Comment (pierDipi) [reply .github/workflows/kind-e2e.yaml:36]: Ok, no problem!

**Comment by matzew** [comment]
LGTM

let's see how this goes..

**Comment by slinkydeveloper** [comment]
This works fine now, probably the problem was the `GO111MODULE` env

**Review by pierDipi — APPROVED** [review]

**Pull request merged**